Mathlib/Analysis/NormedSpace/TrivSqZeroExt.lean

```diff
@@ -176,4 +176,90 @@ end Field
 
 end Topology
 
+/-!
+### The $ℓ^1$ norm on the trivial square zero extension
+
+This is not particularly canonical, but is sufficient to obtain a normed algebra, and thus
+to use `exp_add_of_commute`.
+-/
+
+noncomputable section Normed
+
+section Ring
+
+variable [NormedField 𝕜] [NormedRing R] [NormedAddCommGroup M]
+variable [NormedAlgebra 𝕜 R] [NormedSpace 𝕜 M] [Module R M] [Module Rᵐᵒᵖ M]
+variable [BoundedSMul R M] [BoundedSMul Rᵐᵒᵖ M] [SMulCommClass R Rᵐᵒᵖ M]
+variable [IsScalarTower 𝕜 R M] [IsScalarTower 𝕜 Rᵐᵒᵖ M]
+
+instance instL1NormedAddCommGroup : NormedAddCommGroup (tsze R M) :=
+  inferInstanceAs <| NormedAddCommGroup (WithLp 1 <| R × M)
```

Comment:
Should we be more careful about this and avoid abusing defeq by pulling this back through a composition of equivs? For instance see [Unitization.instUniformSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/Unitization.html#Unitization.instUniformSpace) and [Unitization.instBornology](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/NormedSpace/Unitization.html#Unitization.instBornology).

Also, I know that `WithLp` on products is careful to replace the topology with the product topology, but can you add an `example` verifying this for `TrivSqZeroExt`?

Reply:
I think the defeq abuse is fine here. I would be more open to using the "right" way of doing things if it actually kept the nice defeqs, but without some of the constructions in #8961 I think that's currently not the case (which is why you have to prove a bunch of things to copy the structure, while I don't).